src/shell/layer.rs

```diff
@@ -0,0 +1,110 @@
+/// WLR layer abstraction
+use std::cell::Cell;
+use std::rc::Rc;
+
+use wayland_client::{
+    protocol::{wl_output, wl_surface},
+    Attached, Main,
+};
+
+use wayland_protocols::wlr::unstable::layer_shell::v1::client::{
+    zwlr_layer_shell_v1, zwlr_layer_surface_v1,
+};
+
+pub use wayland_protocols::wlr::unstable::layer_shell::v1::client::{
+    zwlr_layer_shell_v1::Layer, zwlr_layer_surface_v1::Anchor,
+};
+
+/// Render event
+
+#[derive(PartialEq, Copy, Clone)]
+pub enum RenderEvent {
```

Comment:
I'd probably go for `LayerEvent` since `Close` isn't exactly an event related to rendering.